libs/dh/shared/feature-authorization/src/lib/scope.service.spec.ts

```diff
@@ -0,0 +1,224 @@
+
+    const storage = new LocalStorageMock();
+
+    storage.setItem(remainingKey, 'should not be cleared');
+    storage.setItem(scopesKeyToBeRemoved, 'scopes');
+    storage.setItem(scopeKeyToBeRemoved, 'scope');
+
+    // act
+    createTarget(null, true, 'clientId', null, storage);
+
+    // assert
+    expect(storage.getItem(remainingKey)).toBeTruthy();
+    expect(storage.getItem(scopesKeyToBeRemoved)).toBeFalsy();
+    expect(storage.getItem(scopeKeyToBeRemoved)).toBeFalsy();
+  });
+
+  test('should add actor scopes found in claims to local storage', async () => {
+    // arrange
+    const storage = new LocalStorageMock();
+
+    // act
+    createTarget(localAccountId, true, clientId, actorScopesClaim, storage);
+
+    // assert
+    expect(storage.getItem(localAccountId + scopesKey)).toEqual(
+      actorScopesClaim
+    );
+  });
+
+  test.each([true, false])(
+    'getActiveScope should return fallbackValue if account is not found',
+    (granFullAuthorizationEnabled) => {
+      // arrange
+      const target = createTarget(
+        null,
+        granFullAuthorizationEnabled,
+        clientId,
+        null,
+        new LocalStorageMock()
+      );
+
+      // act
+      const actual = target.getActiveScope();
+
+      // assert
+      expect(actual).toEqual(granFullAuthorizationEnabled ? clientId : '');
+    }
+  );
+
+  test.each([true, false])(
+    'getActiveScope should return fallbackValue no scopes are found',
+    (granFullAuthorizationEnabled) => {
+      // arrange
+      const target = createTarget(
+        localAccountId,
+        granFullAuthorizationEnabled,
+        clientId,
+        null,
+        new LocalStorageMock()
+      );
+
+      // act
+      const actual = target.getActiveScope();
+
+      // assert
+      expect(actual).toEqual(granFullAuthorizationEnabled ? clientId : '');
+    }
+  );
+
+  test('getActiveScope returns existing activeScope if contianed in actor scopes from claims', () => {
+    // arrange
+    const storage = new LocalStorageMock();
+
+    const target = createTarget(
+      localAccountId,
+      true,
+      clientId,
+      actorScopesClaim,
+      storage
+    );
+
+    storage.setItem(localAccountId + scopeKey, 'actor2');
+
+    // act
+    const actual = target.getActiveScope();
+
+    // assert
+    expect(actual).toEqual('actor2');
+  });
+
+  test('getActiveScope sets first scope from claims if existing is no longer in claims', () => {
+    // arrange
+    const storage = new LocalStorageMock();
+
+    const target = createTarget(
+      localAccountId,
+      true,
+      clientId,
+      actorScopesClaim,
+      storage
+    );
+
+    storage.setItem(localAccountId + scopeKey, 'actor3');
+
+    // act
+    const actual = target.getActiveScope();
+
+    // assert
+    expect(actual).toEqual('actor1');
+    expect(storage.getItem(localAccountId + scopeKey)).toEqual('actor1');
+  });
+});
+
+class LocalStorageMock {
```

Comment:
suggestion: Looks like [`localStorage`](https://developer.mozilla.org/en-US/docs/Web/API/Window/localStorage) uses `Storage` interface to implement its methods. Let's use the same to make the mock similar to how `localStorage` works.

```suggestion
class LocalStorageMock implements Storage {
```

After adding the interface, TS will complain about missing `length` and `clear` properties.

In order to implement the `length` property, we can turn it into a getter that reads from another internal property. E.g.

```ts
class LocalStorageMock implements Storage {
  store: { [key: string]: string } = {};

  private internalLength = 0;

  public get length() {
    return this.internalLength;
  }

  public readonly setItem = (key: string, value: string) => {
    this.store[key] = value;
    this.internalLength = Object.keys(this.store).length;
  };

  // ...

  public readonly clear = () => {
    this.store = {};
    this.internalLength = 0;
  };
}
```

The `getLength` method should not be part of this mock, since it's not a property on the real `localStorage`.

Reply:
fixed